From 9fab2df10bfed4bb1a4044d8ecac849cbafa058f Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 6 Apr 2024 21:42:05 -0500
Subject: [PATCH] Update __main__.py

---
 scripts/wrap/__main__.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/wrap/__main__.py b/scripts/wrap/__main__.py
index c479cb8cea..46650c14bc 100644
--- a/scripts/wrap/__main__.py
+++ b/scripts/wrap/__main__.py
@@ -1662,8 +1662,8 @@ def fn(name):
                     cpp_files_text = ''
                     for i in cpp_files:
                         cpp_files_text += ' ' + os.path.relpath(i)
-                    libmupdfcpp = f'{build_dirs.dir_so}/libmupdfcpp.so{so_version}'
-                    libmupdf = f'{build_dirs.dir_so}/libmupdf.so{so_version}'
+                    libmupdfcpp = f'{build_dirs.dir_so}/libmupdfcpp.dylib'
+                    libmupdf = f'{build_dirs.dir_so}/libmupdf.dylib'
                     if pyodide:
                         # Compile/link separately. Otherwise
                         # emsdk/upstream/bin/llvm-nm: error: a.out: No such
@@ -1734,8 +1734,8 @@ def fn(name):
                                 command,
                                 force_rebuild,
                                 )
-                        if command_was_run:
-                            macos_patch( libmupdfcpp, f'{build_dirs.dir_so}/libmupdf.dylib{so_version}')
+                        # if command_was_run:
+                        #     macos_patch( libmupdfcpp, f'{build_dirs.dir_so}/libmupdf.dylib{so_version}')
                         if so_version and state.state_.linux:
                             jlib.system(f'ln -sf libmupdfcpp.so{so_version} {build_dirs.dir_so}/libmupdfcpp.so')
 
